webviz_subsurface/plugins/_volumetric_analysis/volume_validator_and_combinator.py

```diff
@@ -0,0 +1,189 @@
+    combined per set. Only region selectors FIPNUM/REGION∕ZONE which are unique whithin each set
+    are kept in the resulting dataframe, if none meets the criteria SET is used.
+
+    Static dataframes must follow a strict format to fit the data processing steps of the
+    InplaceVolumesModel instance. If custom columns found the source is set as dynamic to
+    avoid this processing.
+    Dynamic dataframes must contain a FIPNUM column but custom columns are allowed.
+    """
+
+    ENSEMBLE_COLUMNS = ["ENSEMBLE", "REAL", "SOURCE"]
+    VALID_STATIC_SELECTORS = ["ZONE", "REGION", "FACIES", "LICENSE"]
+    VALID_STATIC_RESPONSES = [
+        "BULK",
+        "NET",
+        "PORV",
+        "HCPV",
+        "GIIP",
+        "STOIIP",
+        "ASSOCIATEDOIL",
+        "ASSOCIATEDGAS",
+    ]
+    FLUID_TYPES = ["TOTAL", "GAS", "OIL"]
+
+    def __init__(self, volumes_table: pd.DataFrame, fipfile: Path = None):
+
+        self.volume_sources: Dict[str, List[str]] = {
+            "static": [],
+            "dynamic": [],
+            "unknown": [],
+        }
+        self.disjoint_set_df = (
+            fipmapper.FipMapper(yamlfile=fipfile).disjoint_sets() if fipfile else None
+        )
+        self.dframe = self.validate_and_combine_sources(volumes_table)
+        self.volume_type = self.set_volumetric_type()
+
+    @property
+    def possible_static_columns(self) -> list:
+        possible_static_columns = self.VALID_STATIC_SELECTORS + self.ENSEMBLE_COLUMNS
+        for fluid in self.FLUID_TYPES:
+            possible_static_columns.extend(
+                [f"{col}_{fluid}" for col in self.VALID_STATIC_RESPONSES]
+            )
+        return possible_static_columns
+
+    def pore_to_porv_mapping(self, dframe: pd.DataFrame) -> pd.DataFrame:
+        """Rename PORE columns to PORV (PORE will be deprecated..)"""
+        return dframe.rename(
+            columns={f"PORE_{fluid}": f"PORV_{fluid}" for fluid in self.FLUID_TYPES},
+            errors="ignore",
+        )
+
+    def validate_and_combine_sources(self, volumes_table: pd.DataFrame) -> pd.DataFrame:
+        """
+        Validate columns for each volumetric source and combine them. Only common columns
+        are kept. If a fipfile is provided volumes are summed per disjoint_set.
+        """
+        dfs = []
+        all_columns = set()
+        for source, voldf in volumes_table.groupby("SOURCE"):
+            voldf = voldf.dropna(axis=1, how="all")
+            voldf = self.pore_to_porv_mapping(voldf)
+            volume_type = self.find_volume_type(voldf.columns, source)
+            self.volume_sources[volume_type].append(source)
+            all_columns.update(voldf.columns)
+            dfs.append(voldf)
+
+        if self.disjoint_set_df is not None:
+            dframe = self.create_set_dframe(volume_dfs=dfs)
+        else:
+            dframe = pd.concat(dfs, join="inner", ignore_index=True)
```

Comment:
```suggestion
        dframe = pd.concat(dfs, join="inner", ignore_index=True) if self.disjoint_set_df is None else self.create_set_dframe(volume_dfs=dfs)
```
...but maybe not more readable :thinking:

Reply:
hehe it is often a bit difficult to find the balance between readability and inliners.. 😉 I will make the the change!